src/Traversable2v.ts

```diff
@@ -297,3 +298,16 @@ export function getTraversableComposition<F, G>(
     }
   }
 }
+
+/**
+ * A traverse followed by flattening the inner result.
+ *
+ * @since 1.14.5
+ */
+export function flatTraverse<F extends URIS3, G extends URIS3>(F: Applicative3<F>, G: Traversable2v3<G> & Chain3<G>): <U, L, A, B>(ta: Type3<G, U, L, A>, f: (a: A) => Type3<F, U, L, Type3<G, U, L, B>>) => Type3<F, U, L, Type3<G, U, L, B>>
+export function flatTraverse<F extends URIS2, G extends URIS2>(F: Applicative2<F>, G: Traversable2v2<G> & Chain2<G>): <L, A, B>(ta: Type2<G, L, A>, f: (a: A) => Type2<F, L, Type2<G, L, B>>) => Type2<F, L, Type2<G, L, B>>
+export function flatTraverse<F extends URIS, G extends URIS>(F: Applicative1<F>, G: Traversable2v1<G> & Chain1<G>): <A, B>(ta: Type<G, A>, f: (a: A) => Type<F, Type<G, B>>) => Type<F, Type<G, B>>
+export function flatTraverse<F, G>(F: Applicative<F>, G: Traversable2v<G> & Chain<G>): <A, B>(ta: HKT<G, A>, f: (a: A) => HKT<F, HKT<G, B>>) => HKT<F, HKT<G, B>>
+export function flatTraverse<F, G>(F: Applicative<F>, G: Traversable2v<G> & Chain<G>): <A, B>(ta: HKT<G, A>, f: (a: A) => HKT<F, HKT<G, B>>) => HKT<F, HKT<G, B>> {
+  return (ta, f) => F.map(G.traverse(F)(ta, f), flatten(G))
```

Comment:
I propose to remove the runtime dependency on the `Chain` module by replacing `flatten` with its expanded counterpart. Also `G.traverse(F)` can be moved outside the returned function as an optimization

```ts
  const traverseF = G.traverse(F)
  return (ta, f) => F.map(traverseF(ta, f), ggb => G.chain(ggb, gb => gb))
```

Reply:
We still need to depend on the `Chain` module to be able to call `G.chain`, no? Or maybe by using the `flatten` function some module optimizations become impossible?

Reply:
Yes but in my version we are only importing types, so there's no runtime dependency. You can double check by inspecting the `lib/Traversable2v.js`file (`var Chain_1 = require("./Chain");` isn't there anymore)